rasa/nlu/utils/bilou_utils.py

```diff
@@ -271,6 +277,32 @@ def ensure_consistent_bilou_tagging(predicted_tags: List[Text]) -> List[Text]:
         if prefix == BEGINNING:
             last_idx = _find_bilou_end(idx, predicted_tags)
 
+            relevant_confidences = predicted_confidences[idx : last_idx + 1]
+            relevant_tags = [
+                tag_without_prefix(tag) for tag in predicted_tags[idx : last_idx + 1]
+            ]
+
+            # if not all tags are the same, for example, B-person I-person L-location
+            # we need to check what tag we should use depending on the confidence
+            # values and update the tags and confidences accordingly
+            if not all(relevant_tags[0] == tag for tag in relevant_tags):
+                avg_confidence_per_tag = _avg_confidence_per_tag(
+                    relevant_tags, relevant_confidences
+                )
+
+                # the entity should have the tag with the highest average confidence
+                tag = max(avg_confidence_per_tag, key=avg_confidence_per_tag.get)
```

Comment:
@Ghostvv do you think this is enough or should we also take into account how many tags within the entity are predicted as e.g. person or location?

Reply:
I think how many tags is more robust measure. E.g. person 2 times with 0.89 and location 1 time with 0.95. Maybe it is possible to come up with some kind of F1 score between these values

Reply:
also, it'd be great to log this mistakes somehow, so that users can be pointed out where more data is required

Reply:
👍

Reply:
updated the metric